presto-product-tests/src/main/java/io/prestosql/tests/hive/TestHiveTransactionalTable.java

```diff
@@ -184,19 +187,66 @@ private static String hiveTableProperties(TransactionalTableType transactionalTa
 
     private static void compactTableAndWait(CompactionMode compactMode, String tableName, String partitionString, Duration timeout)
     {
-        assertEquals(getTableCompactions(compactMode, tableName).count(), 0);
+        log.info("Running %s compaction on %s", compactMode, tableName);
+
+        Failsafe.with(
+                new RetryPolicy<>()
+                        .withMaxDuration(java.time.Duration.ofMillis(timeout.toMillis()))
+                        .withMaxAttempts(Integer.MAX_VALUE))  // limited by MaxDuration
+                .onFailure(event -> {
+                    throw new IllegalStateException(format("Could not compact table %s in %d retries", tableName, event.getAttemptCount()), event.getFailure());
+                })
+                .onSuccess(event -> log.info("Finished %s compaction on %s in %s (%d tries)", compactMode, tableName, event.getElapsedTime(), event.getAttemptCount()))
+                .run(() -> tryCompactingTable(compactMode, tableName, partitionString, Duration.valueOf("90s")));
```

Comment:
is 90s enough? what's typical time it takes on CI?

Reply:
~20-30 seconds on GHA:

MINOR:

```
PT34.838358153S
PT30.844574209S
PT29.576314653S
PT31.662333431S
PT32.115200646S
PT32.932911889S
PT33.893968919S
PT29.637382943S
PT3.384991089S
PT6.792642366S
PT3.277330249S
PT5.378576046S
PT6.337980713S
PT4.7940377S
PT5.384031019S
PT5.475184157S
PT22.72873679S
PT22.162208904S
PT20.25416786S
PT28.962255175S
PT20.369906495S
PT20.36921996S
PT25.370782901S
PT21.177042401S
PT9.567928944S
PT8.48389606S
PT8.445362006S
PT10.563044679S
PT10.54838043S
PT6.392878269S
PT5.366843937S
PT5.411921104S
PT22.50301294S
PT21.209129968S
PT23.706054989S
PT28.109694148S
PT19.413351215S
PT24.236665704S
PT23.882030693S
PT25.434501155S
PT3.470249565S
PT4.540120403S
PT1.292804921S
PT5.375794887S
PT5.472470425S
PT8.654503554S
PT6.425966434S
PT6.408570237S
PT31.023044459S
PT24.549324998S
PT28.908349388S
PT28.839517233S
PT25.091203992S
PT24.729353252S
PT24.728750437S
PT21.367408401S
PT2.676350984S
PT6.433613682S
PT6.800652414S
PT5.835448164S
PT5.359681922S
PT6.436965362S
PT4.375353378S
PT5.390717631S
PT25.930694265S
PT22.174943204S
PT24.251963387S
PT18.069997819S
PT21.13523818S
PT23.32611175S
PT21.484904459S
PT25.693757204S
PT6.402732506S
PT10.529744223S
PT6.753379343S
PT7.800682509S
PT6.482521267S
PT6.772835094S
PT10.488767769S
PT7.495694755S
PT25.937212248S
PT18.261208232S
PT29.76267732S
PT24.525648134S
PT24.777551591S
PT29.647508626S
PT20.467010381S
PT24.691764453S
PT3.290379089S
PT3.306461549S
PT5.762540276S
PT1.651054048S
PT1.289065089S
PT3.740514872S
PT5.395394702S
PT5.374730286S
PT29.95287575S
PT19.395009562S
PT24.513842613S
PT29.871213374S
PT19.313993858S
PT26.507932332S
PT24.39135613S
PT24.543590796S
PT8.915620955S
PT8.524092371S
PT7.836225487S
PT6.478680622S
PT5.522861469S
PT5.800177765S
PT6.480432644S
PT5.85404512S
PT25.69831208S
PT20.698268428S
PT29.107473986S
PT27.216014399S
PT27.000289095S
PT24.051333288S
PT29.803226643S
PT28.584425318S
PT3.316932009S
PT7.48672612S
PT5.401890444S
PT5.630835785S
PT5.355070023S
PT5.42534251S
PT4.353071406S
PT6.516268257S
```

MAJOR:
```
PT34.838358153S
PT30.844574209S
PT29.576314653S
PT31.662333431S
PT32.115200646S
PT32.932911889S
PT33.893968919S
PT29.637382943S
PT3.384991089S
PT6.792642366S
PT3.277330249S
PT5.378576046S
PT6.337980713S
PT4.7940377S
PT5.384031019S
PT5.475184157S
PT22.72873679S
PT22.162208904S
PT20.25416786S
PT28.962255175S
PT20.369906495S
PT20.36921996S
PT25.370782901S
PT21.177042401S
PT9.567928944S
PT8.48389606S
PT8.445362006S
PT10.563044679S
PT10.54838043S
PT6.392878269S
PT5.366843937S
PT5.411921104S
PT22.50301294S
PT21.209129968S
PT23.706054989S
PT28.109694148S
PT19.413351215S
PT24.236665704S
PT23.882030693S
PT25.434501155S
PT3.470249565S
PT4.540120403S
PT1.292804921S
PT5.375794887S
PT5.472470425S
PT8.654503554S
PT6.425966434S
PT6.408570237S
PT31.023044459S
PT24.549324998S
PT28.908349388S
PT28.839517233S
PT25.091203992S
PT24.729353252S
PT24.728750437S
PT21.367408401S
PT2.676350984S
PT6.433613682S
PT6.800652414S
PT5.835448164S
PT5.359681922S
PT6.436965362S
PT4.375353378S
PT5.390717631S
PT25.930694265S
PT22.174943204S
PT24.251963387S
PT18.069997819S
PT21.13523818S
PT23.32611175S
PT21.484904459S
PT25.693757204S
PT6.402732506S
PT10.529744223S
PT6.753379343S
PT7.800682509S
PT6.482521267S
PT6.772835094S
PT10.488767769S
PT7.495694755S
PT25.937212248S
PT18.261208232S
PT29.76267732S
PT24.525648134S
PT24.777551591S
PT29.647508626S
PT20.467010381S
PT24.691764453S
PT3.290379089S
PT3.306461549S
PT5.762540276S
PT1.651054048S
PT1.289065089S
PT3.740514872S
PT5.395394702S
PT5.374730286S
PT29.95287575S
PT19.395009562S
PT24.513842613S
PT29.871213374S
PT19.313993858S
PT26.507932332S
PT24.39135613S
PT24.543590796S
PT8.915620955S
PT8.524092371S
PT7.836225487S
PT6.478680622S
PT5.522861469S
PT5.800177765S
PT6.480432644S
PT5.85404512S
PT25.69831208S
PT20.698268428S
PT29.107473986S
PT27.216014399S
PT27.000289095S
PT24.051333288S
PT29.803226643S
PT28.584425318S
PT3.316932009S
PT7.48672612S
PT5.401890444S
PT5.630835785S
PT5.355070023S
PT5.42534251S
PT4.353071406S
PT6.516268257S

~/Downloads/logs on ☁️  us-east-2
❯ grep 'Finished MAJOR' *.txt | awk '{print $15}'
PT35.491362277S
PT27.040763797S
PT35.31016765S
PT36.626785954S
PT36.653313653S
PT35.163148959S
PT39.572599944S
PT35.06559024S
PT23.29204194S
PT22.201155344S
PT26.901805328S
PT28.072207201S
PT26.915338869S
PT25.916837113S
PT23.751669051S
PT25.962492511S
PT22.141980261S
PT17.177667464S
PT26.184159558S
PT25.635737023S
PT26.720874883S
PT26.237793639S
PT27.078842519S
PT26.006621506S
PT21.297192303S
PT22.391867352S
PT25.822026411S
PT25.632236496S
PT21.89423386S
PT26.494759405S
PT25.551515357S
PT25.109847925S
PT22.197301312S
PT22.376736998S
PT22.586709815S
PT26.586578429S
PT27.712236441S
PT26.955078837S
PT22.787868064S
PT30.576637061S
PT21.46546404S
PT17.084862765S
PT25.873188493S
PT26.453920875S
PT25.179263581S
PT29.925661268S
PT26.802172906S
PT26.463805746S
PT22.791848413S
PT22.346346515S
PT28.961529438S
PT26.614371831S
PT26.398776127S
PT26.839003049S
PT22.701335815S
PT26.656246586S
PT22.696022337S
PT21.443843753S
PT29.963166471S
PT25.706105172S
PT26.305594413S
PT27.54011601S
PT25.764195308S
PT24.92160194S
```